parser/src/ast.rs

```diff
@@ -0,0 +1,92 @@
+use crate::lexer::Token;
+use std::fmt;
+
+/// [Identifier] is used to represent variable names.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Identifier {
+    pub name: String,
+}
+
+impl fmt::Display for Identifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", &self.name)
+    }
+}
+
+/// [Source] is the root node of the AST representing the AIR constraints file.
+#[derive(Debug, PartialEq)]
+pub struct Source(pub Vec<SourceSection>);
+
+/// Source is divided into SourceSections.
+/// There are 4 types of Source Sections:
+/// - AirDef: Name of the air constraints module.
+/// - TraceCols: Trace Columns representing columns of the execution trace.
+/// - BoundaryConstraints: Boundary Constraints to be enforced on the boundaries of columns defined
+///   in the TraceCols section. Currently there are two types of boundaries, First and Last
+///   representing the first and last rows of the column.
+/// - TransitionConstraints: Transition Constraints to be enforced on the trace columns defined
+///   in the TraceCols section.
+#[derive(Debug, PartialEq)]
+pub enum SourceSection {
+    AirDef(AirDef),
+    TraceCols(TraceCols),
+    BoundaryConstraints(BoundaryConstraints),
+    TransitionConstraints(TransitionConstraints),
+}
+
+/// Name of the air constraints module.
+#[derive(Debug, Eq, PartialEq)]
+pub struct AirDef {
+    pub name: Identifier,
+}
+
+/// [TraceCols] contains the trace columns representing columns of the execution trace.
+/// TraceCols is made up of TraceColsGrp which represents groups of columns of a specific type.
```

Comment:
It looks like `TraceColsGrp` has been removed, right? In that case, this comment should be updated.

Reply:
Sorry, fixed.